application/forms/PreferenceForm.php

```diff
@@ -202,6 +203,35 @@ public function createElements(array $formData)
             }
         }
 
+        $regEx = '/@media\s*\(\s*min-height\s*:\s*@prefer-(light|dark)-color-scheme\s*\)\s*,\s*\(\s*prefers-color-scheme\s*:\s*(light|dark)\s*\)\s*and\s*\(\s*min-height\s*:\s*@enable-color-preference\s*\)/';
+        $regEx2 = '/\(\s*min-height\s*:\s*@enable-color-preference\s*\)/';
+        $enableAuto = [];
+        if (isset($formData['theme']) && $formData['theme'] !== StyleSheet::DEFAULT_THEME) {
+            $file = file_get_contents(StyleSheet::getThemeFile($formData['theme']));
+            $hasMode = (bool) preg_match_all($regEx, $file, $matches);
+            $onlyAuto = $hasMode ? null : preg_match($regEx2, $file);
+            $enableAuto = ($hasMode || $onlyAuto) ? ['auto'] : [];
+        }
+
+        $modes = [
+            'dark'      => $this->translate('Dark'),
+            'light'     => $this->translate('Light'),
+            'auto'      => $this->translate('Auto')
+        ];
+
+        $disabled = isset($matches) ? array_diff(array_keys($modes), $matches[1], $enableAuto) : [];
```

Comment:
I guess `$onlyAuto ? ['dark', 'light'] : []` suffices here :wink: 

You could also do that below in the option array.